NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -263,3 +288,146 @@ -(void) run {
+    NSString *path = [bundle pathForResource:@"urls" ofType:@"txt"];
+    
+    mk::ooni::WebConnectivity()
+    .set_options("backend", "https://a.collector.test.ooni.io:4444")
+    .set_options("nameserver", get_dns_server())
+    .set_options("geoip_country_path", [geoip_country UTF8String])
+    .set_options("geoip_asn_path", [geoip_asn UTF8String])
+    .set_options("net/ca_bundle_path", [ca_cert UTF8String])
+    .set_options("save_real_probe_ip", include_ip)
+    .set_options("save_real_probe_asn", include_asn)
+    .set_options("collector_base_url", [collector_address UTF8String])
+    .set_input_filepath([path UTF8String])
+    .set_output_filepath([[self getFileName:@"json"] UTF8String])
+    .set_verbosity(MK_LOG_INFO)
+    .on_log([self](uint32_t type, const char *s) {
+        NSString *current = [NSString stringWithFormat:@"%@: %@", [super getDate],
+                             [NSString stringWithUTF8String:s]];
+        NSLog(@"%s", s);
+        if ((type & MK_LOG_JSON) != 0) {
+            NSData *data = [[NSString stringWithUTF8String:s] dataUsingEncoding:NSUTF8StringEncoding];
+            NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+            self.progress = [[json objectForKey:@"progress"] floatValue];
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [[NSNotificationCenter defaultCenter] postNotificationName:@"reloadTable" object:nil];
+            });
+        }
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self writeOrAppend:current];
+        });
+    })
+    .run([self]() {
+        NSLog(@"web_connectivity testEnded");
+        dispatch_async(dispatch_get_main_queue(), ^{
+            self.completed = TRUE;
+            [TestStorage set_completed:self.test_id];
+            [[NSNotificationCenter defaultCenter] postNotificationName:@"refreshTable" object:self];
+        });
+    });
+}
+
+@end
+
+@implementation NdtTest : NetworkMeasurement
+
+-(id) init {
+    self = [super init];
+    self.name = @"ndt_test";
+    return self;
+}
+
+-(void) run {
+    self.test_id = [NSNumber numberWithDouble:[[NSDate date] timeIntervalSince1970]];
+    self.json_file = [NSString stringWithFormat:@"test-%@.json", self.test_id];
+    self.log_file = [NSString stringWithFormat:@"test-%@.log", self.test_id];
+    [TestStorage add_test:self];
+
+    /*
+    NSBundle *bundle = [NSBundle mainBundle];
+    NSString *geoip_asn = [bundle pathForResource:@"GeoIPASNum" ofType:@"dat"];
+    NSString *geoip_country = [bundle pathForResource:@"GeoIP" ofType:@"dat"];
+    NSString *ca_cert = [bundle pathForResource:@"cacert" ofType:@"pem"];
+    
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory,
+                                                         NSUserDomainMask, YES);
+    NSString *docs_dir = [paths objectAtIndex:0];
+    NSString *ofile = [NSString stringWithFormat:@"%@/output.json", docs_dir];
+    */
```

Comment:
Is this commented out snippet from the code in the other app? In any case, I think we can avoid committing commented out code in this repository.